### PR TITLE
Performance testing

### LINKED ISSUE #67: Out of curiosity: compare performance of azmi with e.g. curl

### PR DESCRIPTION
**Pull request by iricigor**
Added tests and [pipeline](https://dev.azure.com/iiric/azmi/_build?definitionId=29&_a=summary) that is measuring azmi performance. No impact on product.

Resolves #67

### CODE REVIEW

**Comment by iricigor** [comment]
[Performance testing pipeline run](https://dev.azure.com/iiric/azmi/_build/results?buildId=1363&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=c623b9f5-4e91-5154-177e-19d7203ddd0d) based on last commit 9db567c4b89cf744626fa8197689dd1393248451

File: test/performance/1-basic-tests.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+REPEAT=$1
+
+echo 'azmi - build executable'
+dotnet build src/azmi-commandline/azmi-commandline.csproj
```
Comment (MangyCoyote) [inline test/performance/1-basic-tests.sh:6]: dotnet publish ./src/azmi-commandline/azmi-commandline.csproj --configuration Release --self-contained true /p:PublishSingleFile=true --runtime linux-x64
Is what we deliver to users.

File: test/performance/2-single-blob-tests.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+REPEAT=$1
+PREVIOUS_VERSION=$2
+
+echo 'azmi - build executable'
+dotnet build src/azmi-commandline/azmi-commandline.csproj
```
Comment (MangyCoyote) [inline test/performance/2-single-blob-tests.sh:7]: ditto

File: test/performance/1-basic-tests.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+exePath=$(cd ./src/azmi-commandline/bin/Debug/netcoreapp3.0 || exit; pwd)
+PATH="$PATH:$exePath"
```
Comment (MangyCoyote) [inline test/performance/1-basic-tests.sh:9]: I think you want it vice-versa.
If azmi in PATH, recent build won't by used.

File: test/performance/2-single-blob-tests.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+
+printf  "\n=================\n"
+echo "$PREVIOUS_VERSION getblob --blob $BLOB --file download.txt"
+export DOTNET_BUNDLE_EXTRACT_BASE_DIR="$HOME/cache_dotnet_bundle_extract"
+wget --quiet https://azmideb.blob.core.windows.net/azmi-deb/archive/"$PREVIOUS_VERSION"
+chmod +x ./"$PREVIOUS_VERSION"
+./"$PREVIOUS_VERSION" --version
+
+time seq "$REPEAT" | ./"$PREVIOUS_VERSION" getblob --blob $BLOB --file download1.txt --verbose > /dev/null
```
Comment (MangyCoyote) [inline test/performance/2-single-blob-tests.sh:31]: Tests are not same. --verbose .

File: test/performance/1-basic-tests.sh (hunk 3 above)
Comment (MangyCoyote) [reply test/performance/1-basic-tests.sh:9]: Also think of export keyword. I think should be Ok here.

File: test/performance/3-multiple-blobs-tests.sh [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+echo "azmi getblobs - performance testing, repeat count: 1"
+echo "VM size: $(curl -sS -H Metadata:true "http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2018-10-01&format=text")"
+echo "azmi: $(azmi --version)"
+CONT="https://azmitest5.blob.core.windows.net/azmi-pt"
+
+
+
+printf  "\n=================\n"
+echo "azmi listblobs -c $CONT"
```
Comment (MangyCoyote) [inline test/performance/3-multiple-blobs-tests.sh:17]: Counter? Took me 2 secs to realize container.

File: test/performance/3-multiple-blobs-tests.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+time azmi listblobs -c $CONT | wc -l
+sleep 1
```
Comment (MangyCoyote) [inline test/performance/3-multiple-blobs-tests.sh:19]: I think this sleep() operation is not required. Commands are executed sequentially here.

File: test/performance/1-basic-tests.sh (hunk 3 above)
Comment (iricigor) [reply test/performance/1-basic-tests.sh:9]: I will check on `export`

File: test/performance/3-multiple-blobs-tests.sh (hunk 6 above)
Comment (iricigor) [reply test/performance/3-multiple-blobs-tests.sh:19]: Uf, this one was weird. I will add comment to it. Commands are sequential, but it was appearing in pipeline log as they are not. Take a look [here](https://dev.azure.com/iiric/azmi/_build/results?buildId=1362&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=c623b9f5-4e91-5154-177e-19d7203ddd0d&l=44). Adding `sleep 1` fixed it. If you have some other explanation or suggestion, tell me.

File: test/performance/1-basic-tests.sh (hunk 1 above)
Comment (iricigor) [reply test/performance/1-basic-tests.sh]: Fixed in all three scripts.

File: test/performance/2-single-blob-tests.sh (hunk 4 above)
Comment (iricigor) [reply test/performance/2-single-blob-tests.sh]: Fixed

File: test/performance/3-multiple-blobs-tests.sh (hunk 5 above)
Comment (iricigor) [reply test/performance/3-multiple-blobs-tests.sh]: Fixed

**Comment by iricigor** [comment]
Pipeline updated and [run successfully](https://dev.azure.com/iiric/azmi/_build/results?buildId=1374&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a). Pending two consultations: `export` and `sleep 1`.

**Comment by iricigor** [comment]
[New pipeline run](https://dev.azure.com/iiric/azmi/_build/results?buildId=1409&view=results) was successful.

**Pull request merged**